### ISSUE TITLE
attempt to call field 'toggle' (a nil value)

### ISSUE THREAD

**Issue by ExoticMeiZ** [rcarriga/nvim-dap-ui#290]
![dapui](https://github.com/rcarriga/nvim-dap-ui/assets/76839474/f7ac2c72-3d25-4c36-81c0-4f71cbcb61e8)

I've recently run into an issue with dap-ui. I've just started setting it up, and exploring , when I ran into this error. It occurs whenever I try open/close/toggle dap-ui. I had it all working smoothly, and then the next day it threw this error.

Sorry for the lack of context, etc. If anyone has any input, i'd love it. I'm not much of a tech head, so I can get caught on these things sometimes.

Cheers for you time!

**Comment by aikooo7** [comment]
Just want to say that I'm having the same issue, open() and close() works perfect.

**Comment by ExoticMeiZ** [comment]
> Just want to say that I'm having the same issue, open() and close() works perfect.

That's odd, because I am getting the issue with open and close. Could this be a very simple fix, that we're both missing?

**Comment by ExoticMeiZ** [comment]
> Do you actually call the `setup` of `dapui` in your config (yes, I felt really stupid that I didn't):
> 
> ```
> local dapui = require("dapui")
> dapui.setup({})
> ```

Yep, my config is setup correctly. I've tried using a minimal config, etc. It's odd that things are working for others, and not myself. I must have something going on, I'll try to figure it out futher.

**Comment by aikooo7** [comment]
> Do you actually call the `setup` of `dapui` in your config (yes, I felt really stupid that I didn't):
> 
> ```
> local dapui = require("dapui")
> dapui.setup({})
> ```

Since I'm enrolled in all this, I will also respond.

Yes.

Also I was using n Chad when I tried that.

**Comment by rcarriga** [comment]
> I had it all working smoothly, and then the next day it threw this error.

This is almost certainly a local issue. There have been no changes to this repo recently.

The error message makes it seem like whatever way your referencing `dapui` isn't setting it correctly and so it's `nil`

**Comment by baggiponte** [comment]
I had the same issue with `open()` but I added the `dapui.setup({})` and now it works. This is most likely due to me (and @ExoticMeiZ) using lazy.nvim.

**Comment by maiterth** [comment]
I ran into this issue, as when I was reading from a file in the interactive debug session my dap-ui windows closed.
I tried: 
```
vim.api.nvim_set_keymap('n', '<space>T', [[:lua require"dap.ui".toggle()<CR>]], { noremap = true })
```
and saw the error as shown by @ExoticMeiZ . Thougth this is still bugged.
After trying a bit, I realized the following works for me:
```
vim.api.nvim_set_keymap('n', "<space>t", "<cmd>lua require'dapui'.toggle()<cr>", {noremap = true})
```
Note: 
I have one mapped to 'T' the other to 't', the first one throws the error while the second version works.
All my other shortcuts are implemented like the first version, and they work, so definitely something funky going on, but this may help ppl in the meantime. Not to familiar with lua so I didn't dig into it further.

**Comment by ancientjpeg** [comment]
Hi everybody, I encountered this issue and I wanted to give my two cents:

I had initially set my plugin up to track "mfussenegger/nvim-dap" before realizing that was an old fork of this repository. After changing my plugin config to properly track this repository, I got this error as well. After debugging for a bit, I noticed the config table was only partially populated. I deleted/reinstalled the plugin, and it worked fine. I believe that this issue is a result of a stale/corrupt installation of the plugin, and purging it followed by a reinstallation is the fix.
